modules/administration-guide/pages/importing-untrusted-tls-certificates.adoc

```diff
@@ -7,200 +7,180 @@
-====
+{prod-short} uses labeled config maps in {prod-short} {orch-namespace} as sources for TLS certificates.
+The config maps can have an arbitrary amount of keys with a random amount of certificates each.
 
-[IMPORTANT]
+[NOTE]
 ====
-Some {prod-short} components require a full certificate chain to trust the endpoint.
-If the cluster is configured with an intermediate certificate, add the whole chain, including self-signed root, to {prod-short}.
+When an OpenShift cluster contains cluster-wide trusted CA certificates added through the link:https://docs.openshift.com/container-platform/latest/networking/configuring-a-custom-pki.html#nw-proxy-configure-object_configuring-a-custom-pki[cluster-wide-proxy configuration],
+{prod-short} Operator detects them and automatically injects them into a config map with the `config.openshift.io/inject-trusted-cabundle="true"` label.
+Based on this annotation, OpenShift automatically injects the cluster-wide trusted CA certificates inside the `ca-bundle.crt` key of the config map.
 ====
 
-== Adding new CA certificates into {prod-short}
-
-Certificate files are typically stored as Base64 files, with extensions such as `.pem`, `.crt`, `.ca-bundle`, and others. All Secrets that hold certificate files should use the Base64-encoded certificate rather than binary-encoded certificate. The following procedure is applicable for already installed and running instances and for instances that are to be installed.
-
 .Prerequisites
+* An active `{orch-cli}` session with administrative permissions to the destination {orch-name} cluster.
+See {orch-cli-link}.
 
-* An active `{orch-cli}` session with administrative permissions to the destination {orch-name} cluster. See {orch-cli-link}.
-* Namespace for {prod-short} exists.
-* {prod-short} already uses some reserved file names to automatically inject certificates into the `ConfigMap` object. Avoid using the following reserved file names to save your certificates:
-  ** `ca-bundle.crt`
-  ** `ca.crt`
+* The `{prod-namespace}` {orch-namespace} exists.
 
-.Procedure
+* The complete Certificate Authorities chains to import (root CA and intermediate) in link:https://wiki.openssl.org/index.php/PEM[PEM] format in `ca-cert-__<count>__.pem` files.
 
-. Save the certificates you need to import to a local file system.
+.Procedure
+. Concatenate all CA chains PEM files to import, into the `custom-ca-certificates.pem` file, and remove the return character that is incompatible with the Java trust store.
 +
-[CAUTION]
-====
-* A certificate with the introductory phrase `BEGIN TRUSTED CERTIFICATE` is likely in the PEM `TRUSTED CERTIFICATE` format, which is not supported by Java. Convert it to the supported `CERTIFICATE` format with the following command:
-** `openssl x509 -in cert.pem -out cert.cer`
-====
```

Comment:
Replacing carriage return chars addresses the `TRUSTED CERTIFICATE` format not being supported by Java? That would explain why you are removing this "CAUTION" admonition. Otherwise the admonition is still useful in my opinion.

Reply:
Yes. Typically, users will download the CA chain from a browser such as Firefox, and the browser adds the carriage return. Then the certificates are ignored by Java. (I tested with the same CA chain, with and without the carriage return).

The mention about Base64 encoding was a leftover from a previous version of the docs: when you apply a YAML file (rather than referencing a PEM file in the CLI), you must re-encode the PEM certificate in Base64 in the YAML. Applying a YAML file is better for idempotency. But the method proposed now is better for clarity.

Reply:
Additionally, the previous paragraph was:
* pointless, as the command output is a *PEM* certificate. See openssl x509 man page.
* destructive, as the command output is *one* certificate (leaving out the intermediates if any).